### PR TITLE
Prevent decryption of 0RTT connections

### PR DESCRIPTION
**Pull request by awelzel**
When a client starts with a 0RTT, we can not decrypt the server side initial. Prevent from that.

Also, add a quic_history field tracking the different packet types up until a maximum length of 10 for easier debugging.

### CODE REVIEW

**Comment by awelzel** [comment]
@bbannier - mind skimming this? It should fix the majority of analyzer_violations that @initconf has reported.

The script changes aren't nicely split in a separate commit, hope that's okay...

File: analyzer/QUIC.spicy [hunk 1]
```diff
@@ -376,12 +382,21 @@ type Packet = unit(from_client: bool, context: ConnectionIDInfo&) {
...
+    self.long_packet_type = $$.initialbyte.long_packet_type;
   }
 
   # Capture all the packet bytes if we're still have a chance of decrypting the INITIAL PACKETS
-  fpack: AllData &try if ( should_buffer(from_client, context) );
+  switch ( self.header_form == HeaderForm::LONG
+           && self.long_packet_type == LongPacketType::INITIAL
+           && should_buffer(context, from_client) ) {
+
+      # XXX: Could we do this with iterators and random access instead?
```
Comment (bbannier) [inline analyzer/QUIC.spicy:396]: Not sure I follow. As currently written you do not gain anything from a unit switch statement was you still have a single conditional field.

Comment (awelzel) [reply analyzer/QUIC.spicy:396]: Sorry, the XXX and the change away the suffixed `if ( ... )` to the switch are separate.

I think I'll just remove and go back to the suffixed `if` form. I'll remove the XXX, too.

It's mainly that the the suffixed `if` was strange to look and following doesn't work:
```
if ( some_condition ) {
   xs: bytes &size=1;
}
```
So it ended up as it is now, though that's arguably not all that better...
```
switch ( some_condition ) {
  True -> xs: bytes &size=1;
}
```

File: testing/tests/interop/quic-go_quic-go/zerortt.zeek [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# @TEST-REQUIRES: zeek -b -e 'print PacketAnalyzer::ANALYZER_PPP ==  PacketAnalyzer::ANALYZER_PPP'
```
Comment (bbannier) [inline testing/tests/interop/quic-go_quic-go/zerortt.zeek:1]: Similar pattern already exists in `handshake.zeek`, could you document why this is needed? With that I'd know why this test doesn't run with zeek-5.2.

Also for new tests a brief `@TEST-DOC` would be nice (or a comment).

Here and below.

File: testing/tests/vector-max-size-crash.zeek [hunk 3]
```diff
@@ -3,4 +3,4 @@
 # @TEST-EXEC: zeek-cut -m ts uid history service < conn.log > conn.log.cut
 # @TEST-EXEC: zeek-cut -m ts uid cause analyzer_kind analyzer_name failure_reason < analyzer.log > analyzer.log.cut
 # @TEST-EXEC: btest-diff conn.log.cut
-# @TEST-EXEC: TEST_DIFF_CANONIFIER=$SCRIPTS/diff-remove-abspath btest-diff analyzer.log.cut
+# @TEST-EXEC: TEST_DIFF_CANONIFIER=' sed -r "s/\((.+)\.spicy:[0-9]+:[0-9]+\)/(\1.spicy:<line>:<column>)/g" | $SCRIPTS/diff-remove-abspath' btest-diff analyzer.log.cut
```
Comment (bbannier) [inline testing/tests/vector-max-size-crash.zeek:6]: This could be in `diff-remove-abspath` if it indeed sanitizes a path.

Comment (awelzel) [reply testing/tests/vector-max-size-crash.zeek:6]: I'll leave that for when it comes up next.

**Pull request merged**